### PR TITLE
Autospec concept updates

### PR DESCRIPTION
**Pull request by krisdale**
- Get rid of whitespace
- Simplify intro (use training as ref)
- Remove overview heading to be consistent with other pages
- Remove list of control files in preference of existing table of common control files and ref to complete list in readme
- Add Related topics at end of article
- Remove unused overview section incl markers (no references after update to autospec guide)
- Table of example control files updated to show only files to control dependencies (a logical group of control files, rather than a mix)
- Move control files into its own section at end
- Update section How autospec works to include Note bullets in the steps to give picture of complete workflow

Signed-off-by: Kristal Dale <kristal.dale@intel.com>

### CODE REVIEW

**Comment by krisdale** [comment]
@mvincerx please review this PR. If you think anyone else should review, please add.

File: source/clear-linux/concepts/autospec-about.rst [hunk 1]
```diff
@@ -1,112 +1,106 @@
-.. _autospec-about: 
+.. _autospec-about:
 
 Autospec
 ########
 
-.. _incl-autospec-overview:
+A standard RPM build process using ``rpmbuild`` requires a tarball and .spec
+file to start. The .spec provides information about a package, how to
+build it, as well as what should be installed and where.
```
Comment (mvincerx) [inline source/clear-linux/concepts/autospec-about.rst]: Please reverse the order of these two paragraphs. Lines 10-12 are para. #1; lines 6-8 are para. #2.

File: source/clear-linux/concepts/autospec-about.rst [hunk 2]
```diff
@@ -1,112 +1,106 @@
...
-in order to generate a ``spec`` file for you. Although not required, you can 
-influence ``autospec`` by providing control files. 
+How autospec works
+******************
+
+``autospec`` attempts to infer the requirements of the .spec file
+by analyzing the source code and :file:`Makefile` information.
+It will continuously run updated builds based on new information
+discovered from build failures until it has a complete and valid .spec file.
+You may optionally influence the behavior of ``autospec`` by providing
```
Comment (mvincerx) [inline source/clear-linux/concepts/autospec-about.rst]: Please make this a note and remove "optionally." Something like: 
.. note:: 
   You may influence the behavior of ``autospec`` by providing :ref: `control files [...]  See Table 1.

**Comment by krisdale** [comment]
@mvincerx great feedback. I've made revisions and committed. Now ready for review.

File: source/clear-linux/concepts/autospec-about.rst [hunk 3]
```diff
@@ -1,112 +1,104 @@
...
-Whereas a standard RPM build process using ``rpmbuild`` requires a tarball 
-and ``spec`` file to start, ``autospec`` only requires a tarball and package
-name. ``autospec`` analyzes the source code and :file:`Makefile` information 
-in order to generate a ``spec`` file for you. Although not required, you can 
-influence ``autospec`` by providing control files. 
+#. ``autospec`` detects any missed declarations in the .spec.
 
-.. code-block:: console 
+#. If a build error occurs, ``autospec`` stops for user inspection and
+   editing of control files to resolve issues.
```
Comment (phmccarty) [inline source/clear-linux/concepts/autospec-about.rst]: The end goal of autospec is to largely eliminate this step, so autospec will often *automatically* detect the cause of the build failure and restart the build (step 1).

Of course, there are build failures autospec does not detect, and this is where the control files are used, to inform autospec of failure cases it can't detect on its own.

Comment (mvincerx) [reply source/clear-linux/concepts/autospec-about.rst]: @phmccarty. Would you be satisfied with a revision that incorporates your feedback? Please advise: 

Note: However, autospec often detects the cause of the build failure and restarts the build process.

Comment (phmccarty) [reply source/clear-linux/concepts/autospec-about.rst]: @mvincerx I would reword the description. Something like:

_If build errors occur, autospec will scan the build log to try and detect the root cause. If it detects the root cause and knows how to continue, it will restart the build automatically with updated build instructions. Otherwise, it will stop the build for user inspection and editing of control files to resolve the errors._

Comment (mvincerx) [reply source/clear-linux/concepts/autospec-about.rst]: @krisdale. Can you please revise final step with @phmccarty 's recommendation?

Comment (krisdale) [reply source/clear-linux/concepts/autospec-about.rst]: @mvincerx Revised final step(s). @phmccarty please review to confirm it includes the needed details and info.
